homeassistant/components/lamarzocco/calendar.py

```diff
@@ -0,0 +1,101 @@
+"""Calendar platform for La Marzocco espresso machines."""
+from collections.abc import Iterator
+from datetime import datetime, timedelta
+
+from homeassistant.components.calendar import CalendarEntity, CalendarEvent
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import dt as dt_util
+
+from .const import DOMAIN
+from .entity import LaMarzoccoBaseEntity
+
+CALENDAR_KEY = "auto_on_off_schedule"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up switch entities and services."""
+
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities([LaMarzoccoCalendarEntity(coordinator, CALENDAR_KEY)])
+
+
+class LaMarzoccoCalendarEntity(LaMarzoccoBaseEntity, CalendarEntity):
+    """Class representing a La Marzocco calendar."""
+
+    _attr_translation_key = CALENDAR_KEY
+
+    @property
+    def event(self) -> CalendarEvent | None:
+        """Return the next upcoming event."""
```

Comment:
The logic for this function i think should be to return the active or next upcoming event. Is this currently skipping an event if it already started and already ended? It seems sufficient to just skip if it ended, but allow it to be a return value even if it already started and is still active, which represents an `on` state.

Reply:
that's excactly what I was already doing, I removed the unnecessary `and` from the `if`